alacritty_terminal/src/url.rs

```diff
@@ -1,318 +1,40 @@
-            return false;
-        }
-
-        self.advance(cell.c, self.state.len())
+    pub fn contains(&self, point: impl Into<Point<usize>>) -> bool {
+        let point = point.into();
+        point.line <= self.start.line
+            && point.line >= self.end.line
+            && (point.line != self.start.line || point.col > self.start.col)
+            && (point.line != self.end.line || point.col < self.end.col)
     }
 
-    /// Returns the URL if the parser has found any.
-    pub fn url(mut self) -> Option<Url> {
-        // Remove non-alphabetical characters before the scheme
-        // https://tools.ietf.org/html/rfc3986#section-3.1
-        if let Some(index) = self.state.find("://") {
-            let iter =
-                self.state.char_indices().rev().skip_while(|(byte_index, _)| *byte_index >= index);
-            for (byte_index, c) in iter {
-                match c {
-                    'a'..='z' | 'A'..='Z' => (),
-                    _ => {
-                        self.origin =
-                            self.origin.saturating_sub(byte_index + c.width().unwrap_or(1));
-                        self.state = self.state.split_off(byte_index + c.len_utf8());
-                        break;
-                    },
-                }
-            }
-        }
-
-        // Remove non-matching parenthesis and brackets
-        let mut open_parens_count: isize = 0;
-        let mut open_bracks_count: isize = 0;
-        for (i, c) in self.state.char_indices() {
-            match c {
-                '(' => open_parens_count += 1,
-                ')' if open_parens_count > 0 => open_parens_count -= 1,
-                '[' => open_bracks_count += 1,
-                ']' if open_bracks_count > 0 => open_bracks_count -= 1,
-                ')' | ']' => {
-                    self.state.truncate(i);
-                    break;
-                },
-                _ => (),
-            }
-        }
-
-        // Track number of quotes
-        let mut num_quotes = self.state.chars().filter(|&c| c == '\'').count();
-
-        // Remove all characters which aren't allowed at the end of a URL
-        while !self.state.is_empty()
-            && (URL_DENY_END_CHARS.contains(&self.state.chars().last().unwrap())
-                || (num_quotes % 2 != 0 && self.state.ends_with('\''))
-                || self.state.ends_with("''")
-                || self.state.ends_with("()"))
-        {
-            if self.state.pop().unwrap() == '\'' {
-                num_quotes -= 1;
-            }
-        }
-
-        // Check if string is valid url
-        if self.origin > 0 && url::Url::parse(&self.state).is_ok() {
-            for scheme in &URL_SCHEMES {
-                if self.state.starts_with(scheme) {
-                    return Some(Url { origin: self.origin - 1, text: self.state });
-                }
-            }
-        }
-
-        None
-    }
+    pub fn linear_bounds(&self, terminal: &Term) -> RangeInclusive<Linear> {
```

Comment:
@kchibisov This is what I meant in your previous PR by refactoring the `url_bounds_at_point` method. Making this significantly simpler seems to be one advantage of the rfind_url change, besides the obvious advantages when searching for more than just one URL.

I think this is pretty uncontroversial, but thought it might be interesting to see the changes, though if you have any further ideas for improvements, do let me know please.

Reply:
Looks pretty simple. I'll take a look at other changes, but tomorrow (later today).